res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,839 @@
+// lower values make it less, higher value more sensible
+MC7000.jogParams = {
+    jogSensitivity: 30  ,// default: 30
+    maxJogValue: 3      ,// default: 3
+};
+
+/*/////////////////////////////////
+//      USER VARIABLES END       //
+/////////////////////////////////*/
+
+
+/* OTHER VARIABLES - DONT'T TOUCH EXCEPT YOU KNOW WHAT YOU DO */
+
+// Resolution of the jog wheel, set so the spinny
+// Jog LED to match exactly the movement of the Jog Wheel
+// The physical resolution seams to be around 1100
+MC7000.jogWheelTicksPerRevolution = 894;
+
+// Pitch faders up and down values (see above for user input)
+MC7000.posRateRanges = [lowest/100, low/100, middle/100, high/100, highest/100];
+MC7000.negRateRanges = [highest/100, high/100, middle/100, low/100, lowest/100];
+
+// must be "true" for Needle Search to be active 
+MC7000.needleSearchTouched = [true, true, true, true];
+
+// initial value for VINYL mode per Deck (see above for user input)
+MC7000.isVinylMode = [MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn];
+
+// initialize the "factor" function for Spinback
+MC7000.factor = [];
+
+// initialize the PAD Mode to Hot Cue and all others off when starting
+MC7000.PADModeCue = [true, true, true, true];
+MC7000.PADModeCueLoop = [false, false, false, false];
+MC7000.PADModeFlip = [false, false, false, false];
+MC7000.PADModeRoll = [false, false, false, false];
+MC7000.PADModeSavedLoop = [false, false, false, false];
+MC7000.PADModeSlicer = [false, false, false, false];
+MC7000.PADModeSlicerLoop = [false, false, false, false];
+MC7000.PADModeSampler = [false, false, false, false];
+MC7000.PADModeVelSamp = [false, false, false, false];
+MC7000.PADModePitch = [false, false, false, false];
+
+// PAD Mode Colors
+MC7000.padColor = {
+    'alloff': 0x01,         // typically not needed for PADs
+    'hotcueoff': 0x02,      // lightblue Hot Cue inactive
+    'hotcueon': 0x04,       // darkblue Hot Cue active
+    'sampleroff': 0x27,     // light pink Sampler standard colour
+    'samplerloaded': 0x38,  // dark pink Sampler loaded colour
+    'samplerplay': 0x09,    // green Sampler playing
+    'rollon': 0x10,         // BeatloopRoll active colour
+    'rolloff': 0x1B,        // BeatloopRoll off colour
+    'cueloopon': 0x0D,      // Cueloop colour for activated cue point
+    'cueloopoff': 0x1A      // Cueloop colour inactive
+};
+
+/* DECK INITIALIZATION */
+MC7000.init = function () {
+	
+    // Decks
+    MC7000.leftDeck = new MC7000.Deck(1, 3);
+    MC7000.rightDeck = new MC7000.Deck(2, 4);
+    
+    // set default Master Volume to give a little room for mixing
+    engine.setValue("[Master]", "gain", 0.85);
+    
+    // VU meters
+    engine.connectControl("[Channel1]", "VuMeter", "MC7000.VuMeter");
+    engine.connectControl("[Channel2]", "VuMeter", "MC7000.VuMeter");
+    engine.connectControl("[Channel3]", "VuMeter", "MC7000.VuMeter");
+    engine.connectControl("[Channel4]", "VuMeter", "MC7000.VuMeter");
```

Comment:
engine.makeConnection does not work for me, all connected LEDs stay switched off. Changed back to engine.connectControl and all working again.

Reply:
Can you try `engine.makeConnection("[Channel1]", "VuMeter", MC7000.VuMeter).trigger();`?

Reply:
> Can you try `engine.makeConnection("[Channel1]", "VuMeter", MC7000.VuMeter).trigger();`?

tried it without success...

Reply:
From the wiki:
> ### Debugging your mappings
>As mentioned above, you don't have to restart Mixxx, when you're testing your scripts. Every time you save your file, Mixxx will reload it immediately. Additionally if you specify --controllerDebug (or --midiDebug prior to verion 1.11), Mixxx then logs all incoming and outgoing MIDI messages. Also you can use print() in your script to output further messages. The second parameter passed to your init() functions specifies if the controller debug mode is enabled.

Can you try that and post the output? I think if you add the `--developer` flag, the output should also reflect when you create bindings like this.

Reply:
This is what I get on the console output starting MIXXX with --developer or --loglevel trace option.
`Warning [Controller]: "Tried to connect ([Channel1], VuMeter)" to an invalid callback, ignoring.`

Reply:
I think the issue might not be `makeConnection` but rather the callback function. can you insert this just before the `makeConnection` calls:
```javascript
print(typeof MC7000.VuMeter)
```
`Tried to connect (x, y) to an invalid callback, ignoring.` should only occur if `MC7000.VuMeter` is not a function. This could be the fault of the refactor I suggested. Can you see any other errors related to your mapping?

Reply:
> I think the issue might not be `makeConnection` but rather the callback function. can you insert this just before the `makeConnection` calls:
> 
> ```js
> print(typeof MC7000.VuMeter)
> ```
I tried with and without the extension of '.trigger()' but always the same warning message as before. 
`Warning [Controller]: "Tried to connect ([Channel1], VuMeter)" to an invalid callback, ignoring.`

Reply:
Solved it ! Big sorry - my fault. I did overlook the function call is in " " as per engine.connectControl. But I simply overlooked that engine.makeConnection does require the function call without " ". Cheers!

Reply:
exactly. And I think you can omit the `.trigger()` then as well.